src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +144,1336 @@ static DamageTable D_80BBF858 = {
+            this->loadedObjIndex = Object_GetIndex(&play->objectCtx, OBJECT_TAISOU);
+            if (this->loadedObjIndex >= 0) {
+                gSegments[6] = PHYSICAL_TO_VIRTUAL(play->objectCtx.status[this->loadedObjIndex].segment);
+                this->animInfoIndex = EN_S_GORO_ANIM_TAISOU_CHEER;
+                SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+                return true;
+            }
+        }
+    } else if ((this->animInfoIndex == EN_S_GORO_ANIM_TAISOU_CHEER) && !(((EnJg*)this->otherGoron)->flags & 1)) {
+        this->loadedObjIndex = Object_GetIndex(&play->objectCtx, OBJECT_OF1D_MAP);
+        if (this->loadedObjIndex >= 0) {
+            gSegments[6] = PHYSICAL_TO_VIRTUAL(play->objectCtx.status[this->loadedObjIndex].segment);
+            this->animInfoIndex = EN_S_GORO_ANIM_IDLE_STAND;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+            this->skelAnime.curFrame = this->skelAnime.endFrame;
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 EnSGoro_CheckLullaby(EnSGoro* this, PlayState* play) {
+
+    s32 actorType;
+    Player* player = GET_PLAYER(play);
+
+    if ((player->transformation == PLAYER_FORM_GORON) && (play->msgCtx.ocarinaMode == 3)) {
+        if (play->msgCtx.lastPlayedSong == OCARINA_SONG_GORON_LULLABY) {
+            this->animInfoIndex = EN_S_GORO_ANIM_ROLLUP;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+
+            actorType = EN_S_GORO_GET_MAIN_TYPE(&this->actor);
+            this->snorePhase = 0x400 << (actorType + 1);
+            if (!(actorType % 2)) {
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_SNOREPHASE;
+            }
+            this->actionFunc = EnSGoro_SleepAction;
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 EnSGoro_CheckGKBehavior(EnSGoro* this, PlayState* play) {
 
-extern ColliderCylinderInit D_80BBF820;
-extern CollisionCheckInfoInit2 D_80BBF84C;
-extern DamageTable D_80BBF858;
+    s32 actorType;
 
-extern UNK_TYPE D_060091A8;
-extern UNK_TYPE D_06012DE0;
+    if (!EnSGoro_SetPartnerActor_GoronChild(this, play)) {
+        return false;
+    }
+    if ((!(this->actionFlags & EN_S_GORO_ACTIONFLAG_GKQUIET_ACKNOWLEDGED)) &&
+        ((((EnGk*)this->otherGoron)->unk_1E4 & 0x80) || (gSaveContext.save.weekEventReg[22] & 4))) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBCA80.s")
+        this->actionFlags |= EN_S_GORO_ACTIONFLAG_GKQUIET_ACKNOWLEDGED;
+        this->animInfoIndex = EN_S_GORO_ANIM_ROLLUP;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_S_Goro/func_80BBD348.s")
+        actorType = EN_S_GORO_GET_MAIN_TYPE(&this->actor);
+        this->snorePhase = 0x400 << (actorType + 1);
+        if (!(actorType % 2)) {
```

Comment:
```suggestion
        if ((actorType % 2) == 0) {
```
The `!` is more for boolean